src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -117,33 +116,17 @@ object Metric {
     val parser = new DelimitedDataParser(lines=lines, delimiter=Delimiter, ignoreBlankLines=false, trimFields=true)
     val names  = parser.headers.toIndexedSeq
 
-    val clazz             = ReflectionUtil.typeTagToClass[T]
```

Comment:
I really don't like the changes to this method.  
Firstly, it is going to make it considerably slower as it's not generating an intermediate set of tuples for every row it reads from the file.  Secondly I think this is making the major use case (reading back a metrics file) more complicated to fit in with a use case I still don't understand the need for ("reading" metrics from a `Seq[(String,String)]`.

If the latter is the minority use case, why not implement that in terms of the existing `read` method?